### PR TITLE
🌱 improve deploy_observability for Tilt

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR makes tilt_setting options more flexible by allowing to specify which observability tool do you want (vs all or nothing)

### CODE REVIEW

**Comment by sbueringer** [comment]
Tested it locally, works as expected!

Thx!

/lgtm

**Review by killianmuldoon — APPROVED** [review]
Working for me locally

/lgtm

**Review by ykakarap — COMMENTED** [review]
A nice change. 🙂 

Just nits.

File: Tiltfile [hunk 1]
```diff
@@ -322,11 +322,17 @@ def deploy_provider_crds():
     )
 
 def deploy_observability():
-    k8s_yaml(read_file("./.tiltbuild/yaml/observability.tools.yaml"))
+    if "promtail" in settings.get("deploy_observability", []):
+        k8s_yaml(read_file("./.tiltbuild/yaml/promtail.observability.yaml"), allow_duplicates = True)
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply Tiltfile:326]: After installing promtail helm chart I was getting errors due to duplicated objects, this was the suggested fix; quoting the Tilt documentation:

> allow_duplicates ( bool ) – If you try to register the same Kubernetes resource twice, this function will assume this is a mistake and emit an error. Set allow_duplicates=True to allow duplicates. There are some Helm charts that have duplicate resources for esoteric reasons.

😄 for esoteric reasons

**Comment by sbueringer** [comment]
@fabriziopandini 
WDYT can we merge this PR so that https://github.com/kubernetes-sigs/cluster-api/pull/6117 can build on top?

(cc @apricote)

**Comment by apricote** [comment]
There is a slight dependency of `grafana` on `loki`, as the latter is configured as a source in the former. This won't break anything, but the source will still be visible (and broken) in grafana if loki is disabled.

The same situation will occur with Prometheus

File: Tiltfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (apricote) [reply Tiltfile:326]: The `Namespace` resource is duplicated between all three observability tools, so that could explain the issues.

**Comment by fabriziopandini** [comment]
> There is a slight dependency of grafana on loki, as the latter is configured as a source in the former. This won't break > > > > anything, but the source will still be visible (and broken) in grafana if loki is disabled.
> 
> The same situation will occur with Prometheus

That's true, but what I'm trying with this PR is to move us away from all or nothing, because installing too many tools makes the developer flow slow, requires unnecessary resources etc. 
We can eventually iterate later and make dependencies among tools managed in a more developer-friendly way (e.g. erroring if we recognize missing dependency, or automatically adding missing dependencies)

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**